tsfresh/transformers/per_column_imputer.py

```diff
@@ -53,16 +57,30 @@ def fit(self, X, y=None):
             :rtype: Imputer
             """
             if not isinstance(X, pd.DataFrame):
-                X = pd.DataFrame(X.copy())
-
-            if self.col_to_NINF_repl is None or self.col_to_PINF_repl is None or self.col_to_NAN_repl is None:
-                col_to_max, col_to_min, col_to_median = get_range_values_per_column(X)
-                if self.col_to_NINF_repl is None:
-                    self.col_to_NINF_repl = col_to_min
-                if self.col_to_PINF_repl is None:
-                    self.col_to_PINF_repl = col_to_max
-                if self.col_to_NAN_repl is None:
-                    self.col_to_NAN_repl = col_to_median
+                X = pd.DataFrame(X)
+
+            col_to_max, col_to_min, col_to_median = get_range_values_per_column(X)
+
+            if self.col_to_NINF_repl_preset is not None:
+                if not set(X.columns) >= set(self.col_to_NINF_repl_preset.keys()):
```

Comment:
I think it is weird if you provide an custom replacement for a column that is not contained in the feature matrix `X`.

You can always reduce your replacement dictionary to the columns that occur in your data.